database/config.go

```diff
@@ -414,6 +381,17 @@ func (t *indexStore) Get(indexName string) (*IndexConfig, error) {
 	return &idxopts, nil
 }
 
+func (t *indexStore) Replace(cfg IndexConfig) error {
```

Comment:
When replacing something, the API must explicitly state what is the key used to identify the thing we are trying to replace, even if that information is already contained in the value of the second parameter.

```suggestion
func (t *indexStore) Replace(idxName string, cfg IndexConfig) error {
```